### ISSUE TITLE
Missing Examples for user facing function

### ISSUE THREAD

**Issue by jbusecke** [GenericMappingTools/pygmt#1666]
**Description of the desired feature**

First of all, the existing examples are really, really nice! They helped me a lot understanding the functionality.

I was however not able to find examples for `pygmt.Figure.psconvert`, `pygmt.Figure.grd2cpt`, and `pygmt.set_display`.

Furthermore many of the [processing functions](https://www.pygmt.org/latest/api/index.html#data-processing) are not show in the examples as far as I can tell.

**Comment by willschlitzer** [comment]
> **Action point**: I think inline docstring might be the way to go with some of these examples. Upstream GMT has done this (e.g. at https://docs.generic-mapping-tools.org/6.3/basemap.html#examples), so we can probably translate these to PyGMT. The [`pygmt.grdfilter`](https://www.pygmt.org/v0.5.0/api/generated/pygmt.grdfilter.html) function has an example we can use as a reference. Key challenge might be to prevent these inline code examples from running in CI (but definitely check that the syntax is ok).

I agree with using inline examples. I use the GMT inline examples to demonstrate the syntax for non-plotting modules when writing the tests for module-wrapping.

**Comment by willschlitzer** [comment]
@GenericMappingTools/pygmt-maintainers Would it be a good idea to add inline docstring examples (without an associated plot) to all of our module docstrings? That way the page would be a one-stop-shop without needing to go to an example.

**Comment by meghanrjones** [comment]
> @GenericMappingTools/pygmt-maintainers Would it be a good idea to add inline docstring examples (without an associated plot) to all of our module docstrings? That way the page would be a one-stop-shop without needing to go to an example.

Yes to me, as long as we can skip the inline examples in the CI and locally when needed.
